### DISCUSSION TITLE
Hook Form Onchane Validate simultaneously on event ignition

### DISCUSSION THREAD

**Discussion by thunder-fury** [react-hook-form/react-hook-form#6769] [General] [answered]
Hello, Hook Form Team is a person who just tweeted before.
I want to ignite the validate and onChange at the same time.

This is the code I wrote.

↓This is a radio button component
```tsx
<input
   id={`${v.name}_${v.value}`}
   type='radio'
   {...register(name, validate, { onChange })}
   value={v.value}
   defaultChecked={defaultChecked === String(v.value)}
/>
```

```tsx
<input
   id={`${v.name}_${v.value}`}
   type='radio'
   {...register(name,{ validate,  onChange })}
   value={v.value}
   defaultChecked={defaultChecked === String(v.value)}
/>
```

```tsx
<input
   id={`${v.name}_${v.value}`}
   type='radio'
   {...register(name, validate, onChange )}
   value={v.value}
   defaultChecked={defaultChecked === String(v.value)}
/>
```

↓I wrote a component with three patterns written above.
```tsx
<FormRadio
  name='filledName'
  onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
     console.log(e)
  }}
  register={register}
  validate={{ required: 'error messege' }}
  options={[{ id: 1, value: 1, name: 'text1' }, { id: 2, value: 2, name: 'text2' }]}
/>
```

```tsx
 {...register(name, validate, { onChange })}
 {...register(name, validate, onChange )}
```
This is not a validate problem, but onchange does not work.

```tsx
{...register(name,{ validate,  onChange })}
```
This will move onchange, but the validate does not work.

Can you tell me if there is a good way

**Comment by bluebill1049** [comment, marked as answer]
> {...register(name,{ validate,  onChange })}

this is the correct syntax, what do you mean by validation not work?

**Reply by thunder-fury** [reply]
thank you for the reply

> {...register(name,{ validate, onChange })}

 When you coded, this error will come out.

For reference, I am implemented in the NextJS environment.😂
![image](https://user-images.githubusercontent.com/66325822/136890556-7f6424f9-8eb3-4207-b842-43550ec5b5cd.png)

**Reply by bluebill1049** [reply]
can you try to reproduce the issue: https://codesandbox.io/s/competent-lalande-g459k?file=/src/App.js

**Reply by thunder-fury** [reply]
I have discovered Miss in my code in this implementation that I sent you to the codesandbut.💦

I found that Props sent to the Formradio Component I made.
`required: 'error messege' `was a mistake to send to an object.
It was late to notice because it was a variable.😂

In conclusion, I feel like this.↓
```tsx
{...register(name,{ { required: 'error messege' },  onChange })}
```
The conclusion was my Miss

I was really surprised by your fast response and thank you.
